code/game/machinery/scan_gate.dm

```diff
@@ -115,6 +117,19 @@
 				if(istype(I, /obj/item/gun))
 					beep = TRUE
 					break
+		if(SCANGATE_HYGIENE)
+			if(iscarbon(M))
+				var/mob/living/carbon/human/H = M
+				if(H.hygiene <= HYGIENE_LEVEL_DIRTY) //Pungent indeed
```

Comment:
you should have the invert option as well.

We can have High-class areas and "Neet Meet N' Greet" areas

Reply:
Reading code in 2019 hah!